src/game.cpp

```diff
@@ -3178,6 +3178,8 @@ bool game::try_get_right_click_action( action_id &act, const tripoint &mouse_tar
         return false;
     }
 
+    const bool is_adjacent = std::abs(mouse_target.x - u.posx()) <= 1 && std::abs(mouse_target.y - u.posy()) <= 1;
+    const bool is_self = std::abs(mouse_target.x - u.posx()) <= 0 && std::abs(mouse_target.y - u.posy()) <= 0;
```

Comment:
Then we probably need another version of `square_dist()` which can ignore z-coord. I'm absolutely sure it will be used more than once.